libs/market-list/src/lib/components/select-market.tsx

```diff
@@ -0,0 +1,222 @@
+import { useQuery } from '@apollo/client';
+import { t, useDataProvider } from '@vegaprotocol/react-helpers';
+import { Interval } from '@vegaprotocol/types';
+import {
+  Dialog,
+  Intent,
+  Popover,
+  RotatingArrow,
+} from '@vegaprotocol/ui-toolkit';
+import classNames from 'classnames';
+import isNil from 'lodash/isNil';
+import { useMemo, useState } from 'react';
+import { MARKET_LIST_QUERY } from '../markets-data-provider';
+import type { Column } from './select-market-columns';
+import { columns } from './select-market-columns';
+import type { MarketList } from '../__generated__';
+import { useVegaWallet } from '@vegaprotocol/wallet';
+import { positionsDataProvider } from '@vegaprotocol/positions';
+import { mapDataToMarketList } from '../utils/market-list.utils';
+import {
+  SelectMarketTableHeader,
+  SelectMarketTableRow,
+} from './select-market-table';
+
+export const SelectMarketLandingTable = ({
+  data,
+  onSelect,
+}: {
+  data: MarketList | undefined;
+  onSelect: (id: string) => void;
+}) => {
+  const marketList = data && mapDataToMarketList(data);
+  return (
+    <div
+      className="max-h-[40rem] overflow-x-auto"
+      data-testid="select-market-list"
+    >
+      <table className="relative h-full min-w-full whitespace-nowrap">
+        <thead className="sticky top-0 z-10 dark:bg-black bg-white">
+          <SelectMarketTableHeader />
+        </thead>
+        <tbody>
+          {marketList?.map((market, i) => (
+            <SelectMarketTableRow
+              key={i}
+              detailed={false}
+              columns={columns(market, onSelect)}
+            />
+          ))}
+        </tbody>
+      </table>
+    </div>
+  );
+};
+
+export const SelectAllMarketsTableBody = ({
+  data,
+  title = t('All markets'),
+  headers,
+  onSelect,
+}: {
+  data?: MarketList;
+  title?: string;
+  headers?: Column[];
+  onSelect: (id?: string) => void;
+}) => {
+  const marketList = data && mapDataToMarketList(data);
+  return data ? (
```

Comment:
chore (non blocking): Can we just check for marketList here, why check for data?